epochStart/metachain/economics.go

```diff
@@ -140,33 +156,44 @@ func (e *economics) ComputeEndOfEpochEconomics(
 	if err != nil {
 		return nil, err
 	}
-	prevEpochEconomics := prevEpochStart.EpochStart.Economics
 
-	noncesPerShardCurrEpoch, err := e.startNoncePerShardFromLastCrossNotarized(metaBlock.GetNonce(), metaBlock.EpochStart)
-	if err != nil {
-		return nil, err
-	}
+	noncesPerShardCurrEpoch := e.startNoncePerShardFromLastCrossNotarized(metaBlock.GetNonce(), metaBlock.GetEpochStartHandler())
+
+	return &argsComputeEconomics{
+		metaBlock: metaBlockData{
+			epoch:                  metaBlock.Epoch,
```

Comment:
Can you rename this to newEpoch ?

then it will be clearer what is expected.
When executing the metablockv3 that causes this computation, the epoch on that metablock is still the old epoch, but here we need to have the new epoch (+1)